core/controllers/subscriptions.py

```diff
@@ -21,26 +21,51 @@
 from core.controllers import base
 from core.domain import subscription_services
 from core.domain import user_services
+import feconf
 
 
 class SubscribeHandler(base.BaseHandler):
     """Handles operations relating to new subscriptions."""
 
+    POST_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'creator_username': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+    }
+
     @acl_decorators.can_subscribe_to_users
     def post(self):
         creator_id = user_services.get_user_id_from_username(
-            self.payload.get('creator_username'))
+        self.normalized_payload.get('creator_username'))
```

Comment:
Why is this unintended?

Reply:
While reading the code I might have  involuntarily unintended it.Now, I have intended it.